task/sast-coverity-check/0.3/migrations/0.3.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Created for task: sast-coverity-check@0.3
+# Creation time: $(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+declare -r pipeline_file=${1:?missing pipeline file}
+
+# Determine the correct image-digest and image-url values based on task presence
+if yq -e '.spec.tasks[] | select(.name == "build-oci-artifact")' "$pipeline_file" >/dev/null; then
+    image_digest_value="$(yq '.spec.tasks[] | select(.name == "build-oci-artifact") | select(.name == "IMAGE_DIGEST").value' "$pipeline_file")"
+    image_url_value="$(yq '.spec.tasks[] | select(.name == "build-oci-artifact") | select(.name == "IMAGE_URL").value' "$pipeline_file")"
+elif yq -e '.spec.tasks[] | select(.name == "build-image-index")' "$pipeline_file" >/dev/null; then
+    image_digest_value="$(yq '.spec.tasks[] | select(.name == "build-image-index") | select(.name == "IMAGE_DIGEST").value' "$pipeline_file")"
+    image_url_value="$(yq '.spec.tasks[] | select(.name == "build-image-index") | select(.name == "IMAGE_URL").value' "$pipeline_file")"
+else
+    echo "Neither build-oci-artifact nor build-image-index tasks found."
+    exit 1
+fi
```

Comment:
```suggestion
if yq -e '.spec.tasks[] | select(.name == "build-oci-artifact")' "$pipeline_file" >/dev/null; then
    image_digest_value="\$(tasks.build-oci-artifact.results.IMAGE_DIGEST)"
    image_url_value="\$(tasks.build-oci-artifact.results.IMAGE_URL)"
elif yq -e '.spec.tasks[] | select(.name == "build-image-index")' "$pipeline_file" >/dev/null; then
    image_digest_value="\$(tasks.build-image-index.results.IMAGE_DIGEST)"
    image_url_value="\$(tasks.build-image-index.results.IMAGE_URL)"
else
    echo "Neither build-oci-artifact nor build-image-index tasks found."
    exit 0
fi
```

I think these values are what you need. Meanwhile, it's unnecessary to exit with 1 if neither is found. It's actually not an error. It's okay to just log a message properly then terminate the execution normally.

Reply:
I made the fix of the script you suggested , when I run `./hack/validate-migrations.sh ` it trow the below error : 

```
info: prepare Konflux standard pipelines
info: fetch pipeline fbc-builder from bundle quay.io/konflux-ci/tekton-catalog/pipeline-fbc-builder:922d8592c0ac5a5d43ff9b196f49228f87dfcd01 -> /tmp/tmp.Exg7jlCR17-validate-migrations/pipelines/pushed/fbc-builder.yaml
*Warning*: This is an experimental command, it's usage and behavior can change in the next release(s)
info: fetch pipeline docker-build from bundle quay.io/konflux-ci/tekton-catalog/pipeline-docker-build:922d8592c0ac5a5d43ff9b196f49228f87dfcd01 -> /tmp/tmp.Exg7jlCR17-validate-migrations/pipelines/pushed/docker-build.yaml
*Warning*: This is an experimental command, it's usage and behavior can change in the next release(s)
info: fetch pipeline docker-build-oci-ta from bundle quay.io/konflux-ci/tekton-catalog/pipeline-docker-build-oci-ta:922d8592c0ac5a5d43ff9b196f49228f87dfcd01 -> /tmp/tmp.Exg7jlCR17-validate-migrations/pipelines/pushed/docker-build-oci-ta.yaml
*Warning*: This is an experimental command, it's usage and behavior can change in the next release(s)
info: fetch pipeline docker-build-multi-platform-oci-ta from bundle quay.io/konflux-ci/tekton-catalog/pipeline-docker-build-multi-platform-oci-ta:922d8592c0ac5a5d43ff9b196f49228f87dfcd01 -> /tmp/tmp.Exg7jlCR17-validate-migrations/pipelines/pushed/docker-build-multi-platform-oci-ta.yaml
*Warning*: This is an experimental command, it's usage and behavior can change in the next release(s)
info: fetch pipeline maven-zip-build from bundle quay.io/konflux-ci/tekton-catalog/pipeline-maven-zip-build:922d8592c0ac5a5d43ff9b196f49228f87dfcd01 -> /tmp/tmp.Exg7jlCR17-validate-migrations/pipelines/pushed/maven-zip-build.yaml
*Warning*: This is an experimental command, it's usage and behavior can change in the next release(s)
info: fetch pipeline maven-zip-build-oci-ta from bundle quay.io/konflux-ci/tekton-catalog/pipeline-maven-zip-build-oci-ta:922d8592c0ac5a5d43ff9b196f49228f87dfcd01 -> /tmp/tmp.Exg7jlCR17-validate-migrations/pipelines/pushed/maven-zip-build-oci-ta.yaml
*Warning*: This is an experimental command, it's usage and behavior can change in the next release(s)
info: fetch pipeline tekton-bundle-builder from bundle quay.io/konflux-ci/tekton-catalog/pipeline-tekton-bundle-builder:922d8592c0ac5a5d43ff9b196f49228f87dfcd01 -> /tmp/tmp.Exg7jlCR17-validate-migrations/pipelines/pushed/tekton-bundle-builder.yaml
*Warning*: This is an experimental command, it's usage and behavior can change in the next release(s)
Traceback (most recent call last):
  File "<string>", line 3, in <module>
ModuleNotFoundError: No module named 'ruamel'
```

Reply:
python3-ruamel-yaml package is required.

Reply:
so we need to add it to the workflow requirements as part of preparations ?

Reply:
I see that you added to the [script validation-migration.sh ](https://github.com/konflux-ci/build-definitions/blob/ae264d37c028b12e24d32f3326d808cab75d0f79/hack/validate-migration.sh#L53-L68) 
this is a better way printing out the  diffs 
thanks @tkdchen